src/internal/storage/track/postgres_tracker.go

```diff
@@ -115,6 +117,12 @@ func (t *postgresTracker) SetTTLPrefix(ctx context.Context, prefix string, ttl t
 		WHERE str_id LIKE $1 || '%'
 		RETURNING expires_at`, prefix, ttl.Microseconds())
 	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
We should turn this into a `pacherr.NotExist` though, like you did below.

Reply:
Setting the TTL on an empty prefix should still return a nil error regardless of any efficiency changes we make in the layer above.

Reply:
I disagree. We should error if we fail to renew a prefix when we expect there to be tracker objects with that prefix. Relying on the consistency checks that come later will make debugging these issues difficult.

Reply:
Applying anything to a prefix pretty much universally means: "for each element with the prefix, do this", or equivalently "update the elements where they have this prefix".  The only reason this was errorring was because we were trying to get a timestamp using RETURNING.  Not because that was the original behavior by design.

It seems like you want to special case this to make one specific kind of bug more easy to detect, but I don't know why detecting specific renewal edge cases is so important.  They will not impact the integrity of the object graph.  Any real problem with expiring objects will manifest itself in the chunk creation path.

The behavior you're describing relies on additional state in memory to avoid errors, and would prevent multiple processes from calling this on the same prefix, without each process separately handling this error. It might make sense for a method SetTTL.